picocoap/src/coap.c

```diff
@@ -247,13 +247,15 @@ coap_error coap_init_pdu(coap_pdu *pdu)
 	if (pdu->max < 4)
 		return CE_INSUFFICIENT_BUFFER;
 
+	pdu->len = 0;
+	memset(pdu->buf, 0, 4);
```

Comment:
Is that the same 4 that is on L246?  I would make a `COAP_HRD_NO_TOKEN_LEN`  define